### PR TITLE
1387 - Add pagination to providers list

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/LRXpDz2r/1387-add-pagination-to-providers-in-api-support-dashboard

### Changes proposed in this pull request

- Pull in the pagination component from Register and adapt for support API
- Uses Kaminari for now as the component is already wired with it. Later we may need to decouple the display logic from the pagination library methods

Screenshot 2021-04-06 at 14 58 21

### Guidance to review

- Add pagination to providers list

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by darokel** [comment]
[Add pagination to providers in API support dashboard](https://trello.com/c/LRXpDz2r/1387-add-pagination-to-providers-in-api-support-dashboard)

File: app/controllers/support/providers_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module Support
   class ProvidersController < ApplicationController
     def index
-      @pagy, @providers = pagy(Provider.order(:provider_name).includes(:courses, :users))
+      @providers = Provider.order(:provider_name).includes(:courses, :users).page(params[:page] || 1)
```
Comment (defong) [inline app/controllers/support/providers_controller.rb:4]:
```suggestion
      @providers = RecuritmentCycle.current.providers.order(:provider_name).includes(:courses, :users).page(params[:page] || 1)
```

We probably only need the current recruitment cycle, else we may need to display it.

**Review by defong — COMMENTED** [review]
Looking good

File: app/components/paginator/style.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  padding: 0 0 5px 3px;
+}
+
+.moj-pagination__item--prev .moj-pagination__link:before {
+  margin-right: 1px;
+}
+
+.moj-pagination__item--next .moj-pagination__link:after {
+  margin-left: 1px;
+}
```
Comment (defong) [inline app/components/paginator/style.scss:11]: line endings

File: spec/components/paginator/view_spec.rb [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    describe "pagination behaviour" do
+      context "when there is only one page" do
+        it "renders nothing" do
+          expect(rendered_component.text).to eq ""
+        end
+      end
+
+      context "when we are on the first of two pages" do
+        it "renders correct summary message" do
+          expect(rendered_component(current_page: 1, total_count: 29).text).to include "Showing 1 to 25 of 29"
```
Comment (defong) [inline spec/components/paginator/view_spec.rb:35]:
```suggestion
          expect(rendered_component(current_page: 1, total_count: 29).text).to include "Showing 1 to 25 of 29 results"
```

**Review by defong — APPROVED** [review]

File: spec/components/paginator/view_spec.rb (hunk 3 above)
Comment (darokel) [reply spec/components/paginator/view_spec.rb]: done https://github.com/DFE-Digital/teacher-training-api/pull/1846/commits/411546198cee8889733192e1231ac2a633a75c92

File: app/components/paginator/style.scss (hunk 2 above)
Comment (darokel) [reply app/components/paginator/style.scss]: done https://github.com/DFE-Digital/teacher-training-api/pull/1846/commits/411546198cee8889733192e1231ac2a633a75c92

**Pull request merged**